polyfill/lib/ecmascript.mjs

```diff
@@ -1423,14 +1423,35 @@ export const ES = ObjectAssign({}, ES2019, {
     return result;
   },
   ToBigInt: (arg) => {
-    if (bigInt.isInstance(arg)) return arg;
+    if (bigInt.isInstance(arg)) {
+      return arg;
+    }
+
     const prim = ES.ToPrimitive(arg, Number);
-    if (typeof prim === 'number') throw new TypeError('Use BigInt() to convert Number to BigInt');
-    try {
-      return bigInt(prim);
-    } catch (e) {
-      if (e instanceof Error && e.message.startsWith('Invalid integer')) throw new SyntaxError(e.message);
-      throw e;
+    switch (typeof prim) {
+      case 'undefined':
+      case 'object':
+      case 'number':
+      case 'symbol':
+        throw new TypeError(`cannot convert ${typeof arg} to bigint`);
+      case 'string':
+        if (!prim.match(/^\d*$/)) {
```

Comment:
Now this rules out ` ` and `-1`. I would suggest putting the entirety of the allowed syntax from ToBigInt in the regexp:
```suggestion
        if (!prim.match(/^\s*(?:[+-]\d+\s*)?$/)) {
```

Reply:
Sorry, I just noticed https://tc39.es/ecma262/#sec-tonumber-applied-to-the-string-type. Thanks!

Reply:
My bad, it should be `/^s*(?:[+-]?\d+\s*)?$/`

Reply:
I think you mean `/^\s*(?:[+-]?\d+\s*)?$/` :D